nilearn/input_data/nifti_masker.py

```diff
@@ -281,16 +295,23 @@ def fit(self, imgs=None, y=None):
             target_affine=self.target_affine,
             target_shape=self.target_shape,
             copy=False, interpolation='nearest')
-        if self.target_affine is not None:
+        if self.target_affine is not None:  # resample image to target affine
             self.affine_ = self.target_affine
-        else:
+        else:  # resample image to mask affine
             self.affine_ = self.mask_img_.affine
         # Load data in memory
         self.mask_img_.get_data()
         if self.verbose > 10:
             print("[%s.fit] Finished fit" % self.__class__.__name__)
-        if self.reports:  # save inputs for reporting
-            self.input_ = imgs
+
+        if (self.target_shape or self.target_affine) is not None:
```

Comment:
I'm quite surprised by the interaction between the "is" and the "or". Are you sure that it is safe and that it always does what we think that it should?

Reply:
I think it is meant to be

```python
if (self.target_shape is not None) or (self.target_affine is not None)
```
the current line will raise a `ValueError` if either of them is a numpy array